tests/integration/integration.go

```diff
@@ -255,7 +255,7 @@ func K3sStopServer(server *K3sServer) error {
 	if server.log != nil {
 		server.log.Close()
 	}
-	if err := server.cmd.Process.Kill(); err != nil {
+	if err := server.cmd.Process.Signal(syscall.SIGTERM); err != nil {
```

Comment:
Added a 30s wait delay... basically after 30s, if the process still hasn't exited after calling cmd.Wait() it sigkills the process

Reply:
I think you need to `exec.CommandContext` and cancel the context for this to work right, as it relies on either context cancellation or the child process exiting on its own.

From the docs:
```
	// The WaitDelay timer starts when either the associated Context is done or a
	// call to Wait observes that the child process has exited, whichever occurs
	// first. When the delay has elapsed, the command shuts down the child process
	// and/or its I/O pipes.
	//
	// If the child process has failed to exit — perhaps because it ignored or
	// failed to receive a shutdown signal from a Cancel function, or because no
	// Cancel function was set — then it will be terminated using os.Process.Kill.
```